src/sentry/api/endpoints/organization_events_trends.py

```diff
@@ -459,7 +478,8 @@ def get(self, request, organization):
             trend_query.params["aliases"] = self.get_snql_function_aliases(
                 snql_trend_columns, trend_type
             )
-            # Query needs to be resolved after columns because of the aliasing
+            # Both orderby and conditions need to be resolved after the columns because of aliasing
+            trend_query.orderby = trend_query.resolve_orderby(orderby)
```

Comment:
Can this be built into the `TrendsQueryBuilder` somehow? The endpoint shouldn't need to know about the order of resolution if possible.

Reply:
Hmmmm could overwrite the init entirely 🤔 which might be far cleaner in the end even if there's more code repeat. I'll give that a try

Reply:
Going to merge this as is to unblock the upcoming performance landing changes, this can be addressed in a follow-up.